### ISSUE TITLE
Rejects adding valid share also ?

### ISSUE THREAD

**Issue by itstooez** [c00w/bitHopper#547]
I believe when bH adds a reject share to the db, it's also adding a valid share to the db. If I throw a reject from 1 user at a pool the db shows :     diff|1|1|user

In work.py, the def handle_store() will call self.bitHopper.reject_callback() which from data.py adds a reject to the pool & user...then later on in handle_store()...self.bitHopper.data_callback() is called, which in data.py adds a share to the user and pool :/

Can someone more experienced go through the code and verify this ?  We shouldn't be adding a 'share' when it's a 'reject'. I would think.

**Comment by c00w** [comment]
Shares is just total shares. Valid and rejected. This makes our reject percentage be out of total shares as opposed to valid shares. Does the website say valid shares? Then its mislabeled.

**Comment by itstooez** [comment]
Well then isn't the expected payout gathered from all shares...thus counting rejects (that don't get counted by pools) ?

**Comment by c00w** [comment]
Yes thats a good point. We should probably do expected by valid shares not submitted ones. That whole system needs a rewrite since we're duplicating  alot of stuff and its hard to follow. I'll clean it up after I add in the DHT.

**Comment by itstooez** [comment]
Expected payout can be fixed by changing def get_expected_payout() from database.py (pulling rejects in the sql query and then subtracting item[2] from item[0] for shares) to what I put below.  This should push it all the way out to the website and start displaying the proper expected amount for each server and the total at the bottom.  Let me know if that looks good to you, go ahead and push it through if it does :)

```
def get_expected_payout(self, server):
    with self.lock:
        sql = 'select shares, diff, rejects from ' + str(server)
        self.curs.execute(sql)
        result = self.curs.fetchall()
        expected = 0
        for item in result:
            shares = item[0] - item[2]
            difficulty = item[1]
            expected += float(shares)/difficulty * 50
        return expected
```

**Comment by c00w** [comment]
Done, I'm closing this issue.
